github/github.go

```diff
@@ -2,121 +2,193 @@ package github
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
-	"github.com/dgrijalva/jwt-go"
+	"net/url"
+	"strings"
+
 	"github.com/kintohub/utils-go/klog"
 	"github.com/valyala/fasthttp"
-	"strings"
-	"time"
 )
 
 type GithubInterface interface {
-	CallGithub(endpoint, verb, params, authToken string) ([]byte, error)
-	GenerateInstallationToken(installationId string) (string, error)
-	GenBearer(jwtToken string) string
+	GetListRepos(page int32, installationId, userAccessToken string) (*GithubRepositories, error)
+	GetUserInformation(userAccessToken string) (*GithubUserInfo, error)
+	CreateGithubAppToken(installationId string) (string, error)
+	GetUserAccessToken(code string) (string, error)
 }
 
 type github struct {
-	baseUrl       string
-	acceptHeader  string
-	appID         string
-	appPrivateKey []byte
+	appClientID     string
+	appClientSecret string
 }
 
-func New(baseUrl, acceptHeader, appID string, appPrivateKey []byte) GithubInterface {
+var (
+	BASE_API_URL             = "https://api.github.com"
+	BASE_URL                 = "https://github.com"
+	TEMP_ACCEPT_HEADER_VALUE = "application/vnd.github.machine-man-preview+json"
```

Comment:
comment - FYI, for the kinto file, it's `application/vnd.github.v3.raw`

Reply:
ill rename it to `ACCEPT_HEADER_MACHINE_PREVIEW`